test/sequential/crowdloan.API.test.ts

```diff
@@ -0,0 +1,356 @@
+    let leaseStartBlock: number;
+    let leaseEndingBlock: number;
+
+    leaseStartBlock = (await getBlockNumber()) + 2;
+    leaseEndingBlock = (await getBlockNumber()) + 5;
+
+    const userCompleteInitialization = await signTx(
+      api,
+      api.tx.crowdloan.completeInitialization(
+        leaseStartBlock,
+        // @ts-ignore
+        leaseEndingBlock
+      ),
+      testUser1.keyRingPair
+    );
+
+    const eventResponse = getEventResultFromMangataTx(
+      userCompleteInitialization
+    );
+
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("UnknownError");
+
+    leaseStartBlock = (await getBlockNumber()) + 2;
+    leaseEndingBlock = (await getBlockNumber()) + 5;
+
+    const sudoCompleteInitialization = await Sudo.batchAsSudoFinalized(
+      Sudo.sudo(
+        api.tx.crowdloan.completeInitialization(
+          leaseStartBlock,
+          // @ts-ignore
+          leaseEndingBlock
+        )
+      )
+    );
+
+    await waitSudoOperationSuccess(sudoCompleteInitialization);
+
+    crowdloanId = (await api.query.crowdloan.crowdloanId()).toHuman();
+
+    await claimRewards(crowdloanId, testUser1).then((result) => {
+      expect(result.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+  });
+});
+
+test("A reward needs to be fully setup with: setCrowdloanAllocation + initializeRewardVec + completeInitialization", async () => {
+  await setCrowdloanAllocation(crowdloanRewardsAmount);
+
+  crowdloanId = (await api.query.crowdloan.crowdloanId()).toHuman();
+
+  await claimRewards(crowdloanId, testUser1).then((result) => {
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(result.data).toEqual("RewardVecNotFullyInitializedYet");
+  });
+
+  await initializeReward(testUser1, crowdloanRewardsAmount);
+
+  await claimRewards(crowdloanId, testUser1).then((result) => {
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(result.data).toEqual("RewardVecNotFullyInitializedYet");
+  });
+
+  const leaseStartBlock = (await getBlockNumber()) + 2;
+  const leaseEndingBlock = (await getBlockNumber()) + 10;
+
+  await completeInitialization(leaseStartBlock, leaseEndingBlock);
+
+  await claimRewards(crowdloanId, testUser1).then((result) => {
+    expect(result.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+});
+
+test("CL needs to be setup in order", async () => {
+  let leaseStartBlock: number;
+  let leaseEndingBlock: number;
+  let initializationRewards: MangataGenericEvent[];
+  let initializationCrowdloan: MangataGenericEvent[];
+  let BootstrapError: RegistryError;
+
+  initializationRewards = await initializeReward(
+    testUser1,
+    crowdloanRewardsAmount
+  );
+
+  BootstrapError = await getEventErrorFromSudo(
+    initializationRewards.filter(
+      (extrinsicResult) => extrinsicResult.method === "Sudid"
+    )
+  );
+
+  if ((BootstrapError.method = "BatchBeyondFundPot")) {
```

Comment:
plz remove the conditional. A test must be repeatable and concise. We should not accept conditional results

Reply:
issue fixed